CONTRIBUTING.txt

```diff
@@ -511,3 +511,121 @@ Bugs
 ----
 
 Please `report bugs on GitHub <https://github.com/scikit-image/scikit-image/issues>`_.
+
+Benchmarks
+----------
+
+While not mandatory for most pull requests, we ask that performance related
+PRs include a benchmark in order to clearly depict the use-case that is being
+optimized for. In this section we will review how to setup the benchmarks,
+and three commands ``asv dev``, ``asv run`` and ``asv continuous``.
+
+Prerequisites
+~~~~~~~~~~~~~
+Begin by installing `airspeed velocity <https://asv.readthedocs.io/en/stable/>`_
+in your development environment. Prior to installation, be sure to activate your
+development environment, then if using ``venv`` you may install the requirement with::
+
+  source skimage-dev/bin/activate
+  pip install asv
+
+If you are using conda, then the command::
+
+  conda activate skimage-dev
+  conda install asv
+
+is more appropriate. Once installed, it is useful to run the command::
+
+  asv machine
+
+To let airspeed velocity know more information about your machine.
+
+Writing a benchmark
+~~~~~~~~~~~~~~~~~~~
+To write  benchmark, add a file in the ``benchmarks`` directory which contains a
+a class with a ``setup`` method and a ``time_`` method.
+
+The setup method should contain all the code necessary to prepare your data
+before the function you wish to benchmark. The ``time_`` method
+should contain calls to the functions you wish to benchmark.
```

Comment:
I like your structure but I think this part could be more explicit. Maybe something like
```
The ``time_`` method should only contain code you wish to benchmark. 
Therefore it is useful to move everything that prepares the benchmark scenario
into the ``setup`` method. This function is called before calling a ``time_``
method and its execution time is not factored into the benchmarks.
```

Reply:
Thanks!